src/en/developer-layer-example.md

```diff
@@ -0,0 +1,153 @@
+a `layer.yaml` file, which tells the builder what layers and interfaces to
+combine with yours to create the finished charm: includes: ['layer:apache-php',
+'interface:mysql']
+
+### Fill in your charm layer
+
+ With that decided, we're ready to start creating our charm layer. The README,
+icon, and copyright file will be the same. The `metadata.yaml` will be the same
+except that the website relation will be managed for us, so we can leave out
+the provides section:
+name: vanilla
+	summary: Vanilla is an open-source, pluggable, themeable, multi-lingual forum.
+	maintainer: Your Name <your@email.tld>
+	description: |
+  	Vanilla is designed to deploy and grow small communities to scale.
+  	This charm deploys Vanilla Forums as outlined by the Vanilla Forums installation guide.
+	tags:
+	  - social
+	requires:
+	  database:
+	    interface: mysql
+
+
+The apache-php layer requires some additional configuration to tell it how to
+install and configure your application. This goes in an apache.yaml file:
+packages:
+	  - 'php5-mysql'
+	  - 'php5-gd'
+	sites:
+	  vanilla:
+	    options: 'Indexes FollowSymLinks MultiViews'
+	    install_from:
+	      source: https://github.com/vanillaforums/Garden/archive/Vanilla_2.0.18.8.tar.gz#sha256=acf61a7ffca9359c1e1d721777182e51637be59744925935291801ccc8e8fd55
+
+This tells the apache-php layer to install some additional packages for PHP,
+where to fetch your application from (as well as its cryptographic hash to
+verify the payload), and a few options for the vhost entry. The application will
+be installed under /var/www/{site_name} where {site_name} is the name of the
+block under sites (in this case vanilla).
+### Implementing your layer
+
+The apache-php layer and mysql interface use
+[charms.reactive](http://pythonhosted.org/charms.reactive/) to coordinate with
+your layer. This makes it easy to coordinate the state of multiple layers,
+relations, configuration options, etc. Code for reactive handlers should live
+under a reactive/ folder in your charm layer. In our case, we'll use
+`reactive/vanilla.py`. The apache-php layer sets an apache.available state when
+your application is done being installed, and the mysql interface sets a similar
+{relation_name}.available state (where {relation_name} depends on what you
+called the relation in your metadata.yaml) to indicate that MySQL has provided
+you with a complete set of database credentials. Thus, by watching for both of
+those states, you can easily tell when the right time to set up your application
+is:
+
+    @when('apache.available', 'database.available')
+    	def setup_vanilla(mysql):
+    	    pass
+
+You'll notice that the MySQL relation is passed in so that you can easily access
+the database connection information. Since the Apache installation and MySQL
+relation negotiation are all handled for us, the only thing left to do is create
+the conf/config.php file for Vanilla. To make things easy, we will render a
```

Comment:
The convention for files and directories seems to be the code block \`conf/config.php\`